lib/checkautovariables.cpp

```diff
@@ -628,11 +584,26 @@ void CheckAutoVariables::checkVarLifetimeScope(const Token * start, const Token
             } else if (isDeadScope(val.tokvalue->variable()->nameToken(), tok->scope())) {
                 errorInvalidLifetime(tok, &val);
                 break;
-            } else if (tok->variable() && tok->variable()->declarationId() == tok->varId() &&
-                       !tok->variable()->isLocal() && !tok->variable()->isArgument() &&
-                       isInScope(val.tokvalue->variable()->nameToken(), tok->scope())) {
-                errorDanglngLifetime(tok, &val);
-                break;
+            } else if (isInScope(val.tokvalue->variable()->nameToken(), tok->scope())) {
+                const Variable * var = nullptr;
+                const Token * tok2 = tok;
+                if (Token::simpleMatch(tok->astParent(), "=")) {
+                    if (tok->astParent()->astOperand2() == tok) {
+                        var = getLHSVariable(tok->astParent());
+                        tok2 = tok->astParent()->astOperand1();
+                    }
+                }
+                else if (tok->variable() && tok->variable()->declarationId() == tok->varId()) {
+                    var = tok->variable();
+                }
+                else if (Token::simpleMatch(tok->astParent(), "=") && tok->astParent()->astOperand2() == tok) {
```

Comment:
This condition can't be true, because it overlaps the first `if` condition.

Reply:
Actually this is a duplicate from above, so I just removed it.